showcase/showcase.bash

```diff
@@ -21,40 +21,40 @@ if [[ "${BASH_SOURCE[0]}" != "${0}" ]]; then
 	return
 fi
 
-go install ./cmd/protoc-gen-go_gapic
+SHOWCASE_SEMVER=0.7.0
 
-SHOWCASE_SEMVER=0.6.0
-
-pushd showcase
 rm -rf gen
 mkdir gen
 
 curl -L -O https://github.com/googleapis/gapic-showcase/releases/download/v$SHOWCASE_SEMVER/showcase_grpc_service_config.json
 
 protoc \
+	--go_out=plugins=grpc:./gen \
 	--go_gapic_out ./gen \
-	--go_gapic_opt 'go-gapic-package=cloud.google.com/go/showcase/apiv1beta1;showcase' \
+	--go_gapic_opt 'go-gapic-package=github.com/googleapis/gapic-showcase/client;client' \
 	--go_gapic_opt 'grpc-service-config=showcase_grpc_service_config.json' \
 	--descriptor_set_in=<(curl -sSL https://github.com/googleapis/gapic-showcase/releases/download/v$SHOWCASE_SEMVER/gapic-showcase-$SHOWCASE_SEMVER.desc) \
-	google/showcase/v1beta1/echo.proto
+	google/showcase/v1beta1/echo.proto google/showcase/v1beta1/identity.proto
 
-pushd gen/cloud.google.com/go/showcase
-go mod init cloud.google.com/go/showcase
+pushd gen/github.com/googleapis/gapic-showcase
+go mod init github.com/googleapis/gapic-showcase
 popd
 
+go mod edit -replace=github.com/googleapis/gapic-showcase=./gen/github.com/googleapis/gapic-showcase
+
 hostos=$(go env GOHOSTOS)
 hostarch=$(go env GOHOSTARCH)
 
 curl -sSL https://github.com/googleapis/gapic-showcase/releases/download/v$SHOWCASE_SEMVER/gapic-showcase-$SHOWCASE_SEMVER-$hostos-$hostarch.tar.gz | tar xz
 ./gapic-showcase run &
 showcase_pid=$!
 
-stop_showcase() {
+go test -count=1 ./...
+test_exit=$?
+
+cleanup() {
 	kill $showcase_pid
 	# Wait for the process to die, but don't report error from the kill.
-	wait $showcase_pid || true
+	wait $showcase_pid || exit $test_exit
 }
-trap stop_showcase EXIT
-
-go test -count=1 ./...
-popd
+trap cleanup EXIT
```

Comment:
This trap is set, then the script exits, then `cleanup` runs? Can we just run the `cleanup` commands directly?

If we need to exit with the correct code, it might be good to do something like:

```
exit_code=0
cleanup() { ... }
trap cleanup EXIT
go test -count=1 ./...
exit_code=$?
```

The pre-declaration might not even be needed.

Reply:
Done! I moved it above and changed the variable name as you mentioned. It was organized this way before, but the naive bash-er in me thought "oh exit_code needs to be declared to be in scope of `cleanup`!" 🤦‍♂